README.md

```diff
@@ -8,14 +8,157 @@ Based on steverhoades/oauth2-openid-connect-client.
 
 ## Usage
 
-Publish config with artisan command `vender:publish` for provider `LaravelOIDCAuth\OIDCAuthServiceProvider`, and fill it.
+Publish config with artisan command `vender:publish` for provider `LaravelOIDCAuth\OIDCAuthServiceProvider`, and fill
+it.
 
-Use `LaravelOIDCAuth\Authenticate` as auth middleware to redirect directly to OIDC login automatically. Alternatively, you can get login url by yourself and store state string into session storage `oidc-auth.state`, see that middleware.
+Use `LaravelOIDCAuth\Middleware\Authenticate` as auth middleware to redirect directly to OIDC login automatically. Alternatively,
+you can get login url by yourself and store state string into session storage `oidc-auth.state`, see that middleware.
 
 ### `config/auth.php`
 
 Set guard driver to `oidc-auth-session` to remove tokens from session storage on logout.
 
-Set provider driver to `oidc-auth-session` to make `Auth::user()` return a `LaravelOIDCAuth\OIDCUser` authenticable from saved token.
+Set provider driver to `none`, user provider is not suitable.
 
-Alternatively, you can implement a user factory (see `LaravelOIDCAuth\UserFactoryInterface`) for things like creating user DB model and use provider driver like `eloquent`.
+Alternatively, you can implement a user factory (see `LaravelOIDCAuth\Contracts\OIDCAuthenticatableFactory`) for things
+like creating user DB model and use provider driver like `eloquent`.
+See [User Model & User Factory](#user-model--user-factory)
+
+## Documentation
+
+### Tokens
+
+You can use authenticated user object to get tokens.
+
+```php
+// Get login user, returns \LaravelOIDCAuth\Contracts\OIDCAuthenticatable|null
+Auth::user();
+// Get the saved access token, returns \OpenIDConnectClient\AccessToken|null
+Auth::user()->accessToken();
+// Get the saved id token, returns \Lcobucci\JWT\Token|null
+Auth::user()->accessToken()->getIdToken();
+// Get the saved refresh token, returns string|null
+Auth::user()->accessToken()->getRefreshToken();
+```
+
+### ID Token
+
+You can get id token claims like this:
+
+```php
+Route::get('/user', function () {
+    return Auth::user()->accessToken()->getIdToken()->getClaims();
+});
+```
+
+### Refresh the access token
+
+If you are using `oidc-auth-session` guard with `OIDCUser`, you can easily refresh the access token:
+
+```php
+use LaravelOIDCAuth\Facades\OIDC;
+// ...
+Auth::user()->refreshAccessToken();
+```
+
+Typically, you don't need to refresh the access token manually, the following doc shows an automatic way to do this.
```

Comment:
> Typically, you don't need to refresh the access token manually, the following doc shows an automatic way to do this.

You can also turn the `Auto refresh`  feature on instead of refreshing manually.

We should also mention that whether token is actually refreshed by that function if it is not expired. If it is always refreshed, we should document how to check expiration against existing token.

Or even better, `Auto refresh` section should come before this one.

Reply:
I rearranged the doc.